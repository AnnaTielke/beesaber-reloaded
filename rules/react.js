module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
