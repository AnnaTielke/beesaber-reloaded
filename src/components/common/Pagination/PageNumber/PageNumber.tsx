import React from 'react';
import styled from 'styled-components';

import colors from '@/tokens/definitions/color';

const ListElement = styled.li<{ isActive: boolean }>`
  ${({ isActive }) => `
        background-color: ${isActive ? colors.yellow.main : colors.white.light};
        font-size: 1rem;
        font-weight: 900;
        padding: 0.3rem;
        text-align: center;
    `}
`;

const PageLink = styled.a.attrs({ href: '/#' })`
  color: ${colors.page.bgColor.main};
  padding: 0.3rem;
`;

type Props = {
  gotoPage: (page: number) => void;
  isActive: boolean;
  page: number;
};

const PageNumber = ({ isActive, gotoPage, page }: Props): JSX.Element => {
  const handleClick = (e: React.MouseEvent) => {
    e.preventDefault();
    gotoPage(page);
  };

  return (
    <ListElement isActive={isActive}>
      <PageLink onClick={handleClick}>{page}</PageLink>
    </ListElement>
  );
};

export default PageNumber;
