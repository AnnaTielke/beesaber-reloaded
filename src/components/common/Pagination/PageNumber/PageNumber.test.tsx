import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import PageNumber from './PageNumber';

describe('PageNumber', () => {
  const mockGotoPage = jest.fn();

  afterEach(() => jest.resetAllMocks());

  it.each([true, false])(
    'should match the snapshot when isActive = %s',
    (isActive) => {
      const { container } = render(
        <PageNumber gotoPage={mockGotoPage} page={5} isActive={isActive} />
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it('should call gotoPage when clicking the page', () => {
    const page = 5;

    render(<PageNumber gotoPage={mockGotoPage} page={page} isActive={false} />);

    const pageLink = screen.getByRole('link', { name: '5' });

    userEvent.click(pageLink);

    expect(mockGotoPage).toHaveBeenCalledTimes(1);
    expect(mockGotoPage).toHaveBeenCalledWith(page);
  });
});
