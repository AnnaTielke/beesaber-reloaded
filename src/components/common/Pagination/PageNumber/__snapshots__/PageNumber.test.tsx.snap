// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNumber should match the snapshot when isActive = false 1`] = `
.c0 {
  background-color: rgba(255,255,255,0.7);
  font-size: 1rem;
  font-weight: 900;
  padding: 0.3rem;
  text-align: center;
}

.c1 {
  color: #00111f;
  padding: 0.3rem;
}

<li
  class="c0"
>
  <a
    class="c1"
    href="/#"
  >
    5
  </a>
</li>
`;

exports[`PageNumber should match the snapshot when isActive = true 1`] = `
.c0 {
  background-color: #ffeb00;
  font-size: 1rem;
  font-weight: 900;
  padding: 0.3rem;
  text-align: center;
}

.c1 {
  color: #00111f;
  padding: 0.3rem;
}

<li
  class="c0"
>
  <a
    class="c1"
    href="/#"
  >
    5
  </a>
</li>
`;
